### ISSUE TITLE
modifying data (e.g. updateTable) inside a migration

### ISSUE THREAD

**Issue by zuk** [kysely-org/kysely#832]
I need to do some data transformations inside a migration — that is, I want to run `db.updateTable(…).set(…)` inside a migration.

Any pointers on how to do this properly? The docs at https://kysely.dev/docs/migrations mention that

> Migrations can also run normal queries to modify data.

... but I can't find any usage examples of this anywhere else.

The problem I'm running into is that since the Kysely instance available in a migration has `any` as the db inference type, function like\ `.updateTable()` are not properly typed.

For example:

```ts
export async function up(db: Kysely<any>): Promise<void> {
  await db.schema
    .alterTable('emails')
    .addColumn('id', 'text')
    .execute()

  await db
    .updateTable('emails')
    .set(eb => ({ // <-- eb has type `any` here
      id: eb.ref('emails.id') // <-- should I just use `sql.ref('emails.id')` here instead?
    }))
    .where('id', 'is', 'NULL')

  await db.schema
    .alterTable('emails')
    .alterColumn('id', col => col.setNotNull())
})
```

The expression builder `eb` in the `.updateTable('emails').set(…)` has type `any`.

This is not ideal.

Am I doing something wrong, or is there no way around this (e.g. should I be using a fully typed instance of `Kysely` here, or is that discouraged because it makes the code reliant on application code?).

**Comment by koskimas** [comment]
The whole query builder is untyped in the migrations by design. Migrations must always be runnable and the code can never change, but the types change when the project evolves. If we used column types in the migration, what'd happen to your migration when, for example, the `id` field of your `emails` table is renamed to `email_id`?

**Comment by zuk** [comment]
Makes sense, but couldn't the callback to (for example) the untyped `.set((eb) => ...)`  get an instance of `ExpressionBuilder<any>` rather than just `any`?

**Comment by koskimas** [comment]
Oh, it's actually `any` and not `ExpressionBuilder<any>`. Sorry about that. Yeah it should still be an `ExpressionBuilder`.

**Comment by koskimas** [comment]
For now you can use this

```ts
const eb = expressionBuilder<any, string>()
await db
  .updateTable('emails')
  .set({
    id: eb.ref('emails.id')
  })
  .where('id', 'is', null) // <-- null instead of 'NULL'
```

**Comment by koskimas** [comment]
This is now fixed in 0.27.2
